src/Symfony/Component/Messenger/Stamp/LockStamp.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Stamp;
+
+use Symfony\Component\Lock\Key;
+
+final class LockStamp implements StampInterface
+{
+    private Key $key;
+    private ?float $ttl;
+    private bool $shouldBeReleasedBeforeHandlerCall;
+
+    public function __construct(
+        string $key,
```

Comment:
i tend to believe the key should be derived from type+payload

Reply:
I think it can be an improvement later to allow the key to be nullable and autogenerated.

But key should be possible to be passed because you might want/need to use the same key for different payload. For instance let's say you pass the timestamp in the payload but you want to process one message per day, you cannot have the timestamp in the key.

Reply:
fair

but why put a timestamp in the payload, if it's going to be locked by single key anyway ...

Reply:
> but why put a timestamp in the payload, if it's going to be locked by single key anyway ...

You may want to run some computation one time a day, but log the moment of the day the computation was asked for the first time ; dunno. 
Also, some people may want two differents message to share the same key ?

I have some simple usage of this feature (and use it on my project atm, working fine) but I cannot know every potential usage of this stamp, and prefer to start with simplicity and flexibility.

Reply:
simplicity = autogenerating the key by default ;)

Reply:
im considering simplicuty from the consumer/enduser POV

ideally these issue are dealt with now